src/clj/rems/application/search.clj

```diff
@@ -26,14 +27,16 @@
 (def ^:private index-lock (Object.))
 
 (mount/defstate ^Directory search-index
-  :start (let [directory (NIOFSDirectory. (.toPath (io/file (:search-index-path env))))]
+  :start (let [index-dir (.toPath (io/file (:search-index-path env)))]
            (locking index-lock
-             (with-open [writer (IndexWriter. directory (-> (IndexWriterConfig. analyzer)
-                                                            (.setOpenMode IndexWriterConfig$OpenMode/CREATE)))]
-               (.deleteAll writer)))
-           (atom {::directory directory
-                  ::searcher-manager (SearcherManager. directory (SearcherFactory.))
-                  ::last-processed-event-id 0}))
+             ;; delete old index
+             (MoreFiles/deleteDirectoryContents index-dir (into-array [RecursiveDeleteOption/ALLOW_INSECURE]))
+             (let [directory (NIOFSDirectory. index-dir)]
+               ;; create new empty index by creating and closing an indexwriter, otehrwise SearcherManager will fail
+               (.close (IndexWriter. directory (IndexWriterConfig. analyzer)))
```

Comment:
I suppose any exception here will cause the start to fail and leave the files etc. for forensic purposes.

Reply:
Yeah